lib/puppet/module/task.rb

```diff
@@ -61,6 +62,25 @@ def self.is_tasks_filename?(path)
       return true
     end
 
+    def self.find_files(name, metadata)
+      if metadata.key?('files')
+        file_list = metadata['files'].map do |file|
+          splits = file.split(File::SEPARATOR)
+          pup_module = Puppet::Module.find(splits[0])
+          mount = splits[1]
+          unless MOUNTS.include? mount
+            msg = _("Files must be saved in module directories that Puppet makes available via mount points: files, lib, tasks")
```

Comment:
Task files must come from the files, lib or tasks directory of a modules.

We probably should verify this doesn't traverse outside those directories. It could cause weirdness in bolt vs PE.